### PR TITLE
Feature: unreads

### PR DESCRIPTION
**Pull request by ppezaris**

### CODE REVIEW

File: shared/ui/store/users/reducer.ts [hunk 1]
```diff
@@ -154,6 +156,60 @@ export const getPreferences = createSelector(
...
+	(a: any, item: Readable) => item,
+	(umis: UnreadsState | undefined, item: Readable) => {
+		if (!umis || !item) return false;
+		const { lastReadItems } = umis;
+		return isItemUnread(item, lastReadItems[item.id]);
+	}
+);
+
+export const unreadMap = createSelector(
+	(state: CodeStreamState) => (isFeatureEnabled(state, "readItem2") ? state.umis : undefined),
```
Comment (cstryker) [inline shared/ui/store/users/reducer.ts:194]: what's this?

File: shared/ui/Stream/Codemark.tsx [hunk 2]
```diff
@@ -2109,6 +2124,8 @@ const mapStateToProps = (state: CodeStreamState, props: InheritedProps): Connect
 			? getReview(state.reviews, codemark.reviewId)
 			: undefined;
 
+	const unread = isUnread(state, codemark!);
+	console.warn("UNREAD IS: ", unread, " for codemark ", codemark);
```
Comment (cstryker) [inline shared/ui/Stream/Codemark.tsx:2128]: remove warn?

File: shared/ui/Stream/Codemark.tsx [hunk 3]
```diff
@@ -160,6 +160,9 @@ interface InheritedProps {
 
 type Props = InheritedProps & DispatchProps & ConnectedProps;
 
+const EMPTY_OBJECT = {};
+const EMPTY_OBJECT2 = {};
```
Comment (cstryker) [inline shared/ui/Stream/Codemark.tsx:164]: what's this for?

File: shared/ui/utils.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ export interface Disposable {
 	dispose(): void;
 }
 
+// DO NOT USE THESE
```
Comment (cstryker) [inline shared/ui/utils.ts:10]: not that i would, but why?

**Comment by cstryker** [comment]
other than minor comments, LGTM

File: shared/ui/utils.ts (hunk 4 above)
Comment (ppezaris) [reply shared/ui/utils.ts:10]: because there global, and if they're used in multiple places you might have unintended consequences if they get set.

**Pull request merged**